### PR TITLE
feat: built-in OPA Rego policy-based configuration audit scanner

### PR DESCRIPTION
**Pull request by danielpacak**
1. Perform synchronous configuration audit within reconciliation loop instead of
   creating Kubernetes Job objects and associated Secrets.
2. Configuration audits at scale with build-in controller-runtime workers.
3. Support NetworkPolicies, PodSecurityPolicies, Ingress, ResourceQuota, and
   possibly other resources.
4. Use OPA Go SDK to evaluate policies read from the `starboard-policy-config`
   ConfigMap, which is compatible with Conftest plugin and KubeEnforcer.
   This is just an MVP, but for production workloads OPA policies can be
   prepared for evaluation and cached similarly to what OPA server does.
5. Preinstall all Kubernetes policies from
   https://github.com/aquasecurity/defsec/tree/v0.14.0/rules/kubernetes
6. First step to deprecate and remove Polaris and Conftest plugins to streamline
   code, usages, and documentation.

Resolves: #889

Signed-off-by: Daniel Pacak <pacak.daniel@gmail.com>

### CODE REVIEW

File: pkg/policy/policy.go [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+// NewMetadata constructs new Metadata based on raw value.
+func NewMetadata(value map[string]interface{}) (Metadata, error) {
+	severity, err := v1alpha1.StringToSeverity(fmt.Sprintf("%v", value["severity"]))
+	if err != nil {
+		return Metadata{}, fmt.Errorf("failed parsing severity from raw value: %w", err)
+	}
+	return Metadata{
+		Severity:    severity,
+		ID:          fmt.Sprintf("%v", value["id"]),
```
Comment (chen-keinan) [inline pkg/policy/policy.go:45]: value["id"] is not checked for existence , id can be set for "<nil>"

File: pkg/policy/policy.go [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+	Title    string
+	Severity v1alpha1.Severity
+	Type     string
+
+	Message string
+}
+
+// NewResult constructs new Result based on raw value.
+func NewResult(value map[string]interface{}) Result {
+	return Result{
```
Comment (chen-keinan) [inline pkg/policy/policy.go:64]: same here value["id"] should be checked for existance

File: pkg/configauditreport/controller.go [hunk 3]
```diff
@@ -0,0 +1,437 @@
...
+		return v1alpha1.ConfigAuditReportData{}, err
+	}
+
+	summary := v1alpha1.ConfigAuditSummary{}
+
+	for _, check := range checks {
+		if check.Success {
+			continue
+		}
+		switch check.Severity {
```
Comment (chen-keinan) [inline pkg/configauditreport/controller.go:283]: I would export the severity switch  check to a util , is it commonly used , we have a similar code in :
https://github.com/aquasecurity/starboard/blob/main/pkg/plugin/polaris/plugin.go#L522

File: pkg/operator/etc/config.go [hunk 4]
```diff
@@ -25,15 +25,31 @@ type Config struct {
...
+	LeaderElectionID      string `env:"OPERATOR_LEADER_ELECTION_ID" envDefault:"starboard-lock"`
 }
 
 // GetOperatorConfig loads Config from environment variables.
 func GetOperatorConfig() (Config, error) {
 	var config Config
 	err := env.Parse(&config)
+
+	if config.ConfigAuditScannerEnabled && config.ConfigAuditScannerBuiltIn {
+		return Config{}, fmt.Errorf("you cannot enabled both plugin-based and built-in configuration audit scanner")
```
Comment (chen-keinan) [inline pkg/operator/etc/config.go:50]: maybe replace with objective message  `plugin-based and built-in configuration audit scanners cannot be enabled at the same time`

File: pkg/policy/policy.go (hunk 1 above)
Comment (danielpacak) [reply pkg/policy/policy.go]: Good point. I added checks for nil and blank strings for mandatory properties.

File: pkg/policy/policy.go (hunk 2 above)
Comment (danielpacak) [reply pkg/policy/policy.go]: Done

File: pkg/configauditreport/controller.go (hunk 3 above)
Comment (danielpacak) [reply pkg/configauditreport/controller.go]: I've extracted this logic to the `v1alpha1.ConfigAuditSummaryFromChecks(checks []v1alpha1.Check)` method.

File: pkg/operator/etc/config.go (hunk 4 above)
Comment (danielpacak) [reply pkg/operator/etc/config.go]: Done

**Review by chen-keinan — APPROVED** [review]

**Pull request merged**